pandas/core/groupby.py

```diff
@@ -1329,6 +1329,27 @@ def cumsum(self, axis=0):
 
     @Substitution(name='groupby')
     @Appender(_doc_template)
+    def rank(self, axis=0, method='average', numeric_only=True,
+             na_option='keep', ascending=True, pct=False):
+        """Compute numerical data ranks (1 through n) along axis.
+        """
+
+        if numeric_only:
+            data = self._obj_with_exclusions._get_numeric_data()
+            if data.size == 0:
+                raise DataError('No numeric types to aggregate')
+            data = data.groupby(self.grouper)
+        else:
```

Comment:
There is still something I don't understand, but maybe that's because I'm not familiar enough with how pandas works internally.

Correct me, but from what I've understood the following piece of code should be useless
```
data = self._obj_with_exclusions._get_numeric_data()
data = data.groupby(self.grouper)
```
because at that point we should already have numeric values only.

And yet, with
```
dg = DataFrame({'a': ['A1', 'A1', 'A1'], 'b': ['B1', 'B1', 'B2'], 'c': 1.}).set_index('a').groupby('c')
```
that I use in the tests, if I add the following prints inside the method
```
print 'self', self._obj_with_exclusions
print 'data', data._obj_with_exclusions
```
I get
```
-------------------- >> begin captured stdout << ---------------------
self      b
a     
A1  B1
A1  B1
A1  B2
data Empty DataFrame
Columns: []
Index: [A1, A1, A1]
--------------------- >> end captured stdout << ----------------------
```
So now I'm really confused :S